OpenRA.Platforms.Default/OpenAlSoundEngine.cs

```diff
@@ -56,19 +56,25 @@ static string[] QueryDevices(string label, int type)
 
 			var devices = new List<string>();
 			var next = ALC10.alcGetString(IntPtr.Zero, type);
+			if (next == IntPtr.Zero || AL10.alGetError() != AL10.AL_NO_ERROR)
+			{
+				Log.Write("sound", "Failed to query OpenAL device list using {0}", label);
+				return new string[] { };
+			}
+
 			do
 			{
 				var str = Marshal.PtrToStringAnsi(next);
+				if (string.IsNullOrEmpty(str))
```

Comment:
This condition will never be true.  As mentioned earlier `PtrToStringAnsi` will only return null if `next` is null, and that case is covered by the check above.  `str` can also never be an empty string, because we are explicitly checking for that as our loop-termination condition.